scapy/contrib/automotive/ecu.py

```diff
@@ -19,7 +19,51 @@
 from scapy.sessions import DefaultSession
 from scapy.ansmachine import AnsweringMachine
 
-__all__ = ["ECU", "ECUResponse", "ECUSession", "ECU_am"]
+__all__ = ["ECU_State", "ECU", "ECUResponse", "ECUSession", "ECU_am"]
+
+
+class ECU_State(object):
+    def __init__(self, session=1, tester_present=False, security_level=0,
+                 communication_control=0, **kwargs):
+        self.session = session
+        self.security_level = security_level
+        self.communication_control = communication_control
+        self._tp = tester_present
+        self.misc = kwargs
+
+    def reset(self):
+        self.session = 1
+        self.security_level = 0
+        self.communication_control = 0
+        self._tp = False
+        self.misc = dict()
+
+    @property
+    def tp(self):
+        return self._tp or self.session > 1
+
+    def __eq__(self, other):
+        return other.session == self.session and other.tp == self.tp and \
+            other.misc == self.misc and \
+            self.security_level == other.security_level
+
+    def __ne__(self, other):
+        return not other == self
+
+    def __lt__(self, other):
+        if self.session < other.session:
+            return True
+        if self.session == other.session:
+            return len(self.misc.keys()) < len(other.misc.keys())
+        return False
+
```

Comment:
this could be changed to:

```python
        if self.session == other.session:
            return len(self.misc) < len(other.misc)
        return self.session < other.session
```
Note the removal of `.keys()` twice.

Reply:
Thanks